lib/configurator/index.js

```diff
@@ -0,0 +1,48 @@
+const EventEmitter = require('events');
+const hash = require('crypto').createHash;
+const storesFactory = require('./stores');
+
+class Configurator extends EventEmitter {
+  constructor(config) {
+    super();
+    const type = config.type;
+    this.logger = config.logger;
+    this.metrics = config.metrics;
+
+    this.store = config.store || new storesFactory(type, config);
+
+    this.pollInterval = config.interval || 300000; // 5 minutes
+    this.currentHash = hash('sha1').update(String(config.currentVal)).digest('base64');
+    this.startPolling();
+  }
+
+  checkForChanges() {
+    this.logger.info('Configurator: fetching config');
+    const start = Date.now();
+    this.store.fetch((err, data) => {
+      if (err) {
+        return this.logger.error(err);
+      }
+
+      this.metrics.histogram('configurator.fetch', Date.now() - start);
+      const newHash = hash('sha1').update(data).digest('base64');
+
+      if (newHash !== this.currentHash) {
+        this.emit('changed', data);
+        this.currentHash = newHash;
+      }
+    });
+  }
+
+  startPolling() {
```

Comment:
This is a minor thing, feel free to ignore, but to me the store should notify you of changes and the configurator should be agnostic to the method used to check if there are changes: some store might use polling, some other might use something else, it depends on the store capacities.